vpd-tool/src/vpd_tool.cpp

```diff
@@ -8,13 +14,60 @@ int VpdTool::readKeyword(const std::string& i_fruPath,
                          const bool i_onHardware,
                          const std::string& i_fileToSave)
 {
-    // ToDo: Need to add implementation details
-    (void)i_fruPath;
-    (void)i_recordName;
-    (void)i_keywordName;
-    (void)i_onHardware;
-    (void)i_fileToSave;
-
-    return 0;
+    int l_rc = -1;
+    try
+    {
+        types::DbusVariantType l_keywordValue;
+        if (i_onHardware)
+        {
+            // TODO: Implement read keyword's value from hardware
+        }
+        else
+        {
+            std::string l_inventoryObjectPath(constants::baseInventoryPath +
+                                              i_fruPath);
```

Comment:
what if the user provides eeprom path without giving -H option ? In that case Line 28 binds the given eeprom path with inventory prefix which will fail in next step.
please handle it

Reply:
if user providing eeprom path without choosing -H option, 
then this request is considered as dbus request.

user has to apply -H.